### PR TITLE
feat(bpdm): add error handling for fetch legal entity 

### PR DESCRIPTION
**Pull request by Phil91**
## Description

Changed the url to fetch the legal entity, for the bpn check in the application process

## Why

The endpoint got enhanced by an errorMessage for the internal error handling. With the new endpoint an operator is able to see the specific error that occured and handle it.

## Issue

N/A - Jira Issue: CPLP-2143

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/externalsystems/Bpdm.Library/BpdmService.cs [hunk 1]
```diff
@@ -122,27 +121,41 @@ public async Task<bool> PutInputLegalEntity(BpdmTransferData data, CancellationT
...
-                _options,
+                Options,
                 cancellationToken: cancellationToken).ConfigureAwait(false);
-            if (legalEntityResponse?.ExternalId == null)
+            if (paginationResponse == null || (paginationResponse.Content == null && paginationResponse.Errors == null))
+            {
+                throw new ServiceException("Access to external system bpdm did not return a valid legal entity response", true);
+            }
+
+            if (paginationResponse.Errors.Any())
```
Comment (ntruchsess) [inline src/externalsystems/Bpdm.Library/BpdmService.cs:148]: use 'IfAny()':
```
            paginationResponse.Errors.IfAny(errors =>
                throw new ServiceException($"The external system bpdm responded with errors {string.Join(";", errors.Select(x => $"ErrorCode: {x.ErrorCode}, ErrorMessage: {x.Message}"))}"));
```

File: src/externalsystems/Bpdm.Library/BpdmService.cs [hunk 2]
```diff
@@ -122,27 +121,41 @@ public async Task<bool> PutInputLegalEntity(BpdmTransferData data, CancellationT
...
         try
         {
-            var legalEntityResponse = await JsonSerializer.DeserializeAsync<BpdmLegalEntityData>(
+            var paginationResponse = await JsonSerializer.DeserializeAsync<PageOutputResponseBpdmLegalEntityData>(
                 responseStream,
-                _options,
+                Options,
                 cancellationToken: cancellationToken).ConfigureAwait(false);
-            if (legalEntityResponse?.ExternalId == null)
+            if (paginationResponse == null || (paginationResponse.Content == null && paginationResponse.Errors == null))
```
Comment (ntruchsess) [inline src/externalsystems/Bpdm.Library/BpdmService.cs:143]: the check of Content and Errors seems to be redundant as both are defined as 'non-nullable' in BpdmLegalEntityOutputData.cs

File: src/externalsystems/Bpdm.Library/BpdmService.cs [hunk 3]
```diff
@@ -122,27 +121,41 @@ public async Task<bool> PutInputLegalEntity(BpdmTransferData data, CancellationT
...
+                throw new ServiceException("Access to external system bpdm did not return a valid legal entity response", true);
+            }
+
+            if (paginationResponse.Errors.Any())
             {
-                throw new ServiceException("Access to external system bpdm did not return a valid legal entity response");
+                throw new ServiceException($"The external system bpdm responded with errors {string.Join(";", paginationResponse.Errors.Select(x => $"ErrorCode: {x.ErrorCode}, ErrorMessage: {x.Message}"))}");
             }
+
+            if (paginationResponse.Content!.Count(x => x.ExternalId == externalId) != 1)
```
Comment (ntruchsess) [inline src/externalsystems/Bpdm.Library/BpdmService.cs:153]: why not:
```
            try
            {
                return paginationResponse.Content!.Single(x => x.ExternalId == externalId);
            }
            catch(InvalidOperationException)
            {
                throw new ServiceException("Access to external system bpdm did not return a valid legal entity response", true);
            }
```

File: src/externalsystems/Bpdm.Library/BusinessLogic/IBpdmBusinessLogic.cs [hunk 4]
```diff
@@ -18,6 +18,7 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
```
Comment (ntruchsess) [inline src/externalsystems/Bpdm.Library/BusinessLogic/IBpdmBusinessLogic.cs:21]: using Enums is not referenced and can be removed

File: src/externalsystems/Bpdm.Library/IBpdmService.cs [hunk 5]
```diff
@@ -19,6 +19,8 @@
  ********************************************************************************/
 
 using Org.Eclipse.TractusX.Portal.Backend.Bpdm.Library.Models;
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
```
Comment (ntruchsess) [inline src/externalsystems/Bpdm.Library/IBpdmService.cs:22]: both usings (Enums + Library) are not referenced and shall be removed

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/externalsystems/Bpdm.Library/BpdmService.cs (hunk 2 above)
Comment (Phil91) [reply src/externalsystems/Bpdm.Library/BpdmService.cs]: the problem is that somehow the json serializer completly ignores the fact that the two values are not nullable and sets them to null. there is now jsonException thrown or anything. I therefor decided to mark the two properties as nullable and do the check.

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**